### PR TITLE
TK-05758 TK-05863 push to bootstrap

### PR DESCRIPTION
**Pull request by thedavidmeister**

### CODE REVIEW

File: crates/kitsune_p2p/kitsune_p2p/src/types/agent_store.rs [hunk 1]
```diff
@@ -12,21 +12,26 @@ pub type Urls = Vec<Url2>;
 /// Value in the peer database that tracks an Agent's representation as signed by that agent.
 #[derive(serde::Serialize, serde::Deserialize, Clone, Debug, PartialEq, derive_more::AsRef)]
 pub struct AgentInfoSigned {
```
Comment (thedavidmeister) [inline crates/kitsune_p2p/kitsune_p2p/src/types/agent_store.rs:14]: note the changes here that make signed info match what the bootstrap expects

importantly the _serialized_ agent info data is on this struct as the _exactly bytes that were signed_ alongside the pubkey for the signature, to allow for context-free, unambiguous signature checking _before_ any deserialization is attempted (e.g. we don't rely on messagepack conventions to determine bytes as a validator)

File: crates/kitsune_p2p/kitsune_p2p/src/types.rs [hunk 2]
```diff
@@ -31,6 +31,10 @@ pub enum KitsuneP2pError {
     /// Other
     #[error("Other: {0}")]
     Other(Box<dyn std::error::Error + Send + Sync>),
+
+    /// Reqwest crate.
+    #[error(transparent)]
+    Reqwest(#[from] reqwest::Error),
```
Comment (neonphog) [inline crates/kitsune_p2p/kitsune_p2p/src/types.rs:37]:
```suggestion
    /// Reqwest crate.
    #[error(transparent)]
    Reqwest(#[from] reqwest::Error),
    
    /// Other
    #[error("Other: {0}")]
    Other(Box<dyn std::error::Error + Send + Sync>),
```

just... personal preference to keep the catch-all generic error type at the bottom : )

File: crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/space.rs [hunk 3]
```diff
@@ -250,20 +251,25 @@ impl SpaceInternalHandler for Space {
...
                 };
                 let sig = evt_sender.sign_network_data(sign_req).await?;
-                let agent_info_signed =
-                    crate::types::agent_store::AgentInfoSigned::try_new(sig, agent_info)?;
+                let agent_info_signed = crate::types::agent_store::AgentInfoSigned::try_new(
+                    (*agent).clone(),
+                    sig,
+                    data,
+                )?;
                 tracing::debug!(?agent_info_signed);
```
Comment (neonphog) [inline crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/space.rs:262]:
```suggestion
                tracing::debug!(?agent_info, ?sig);
```

?? - not sure the best thing to debug now, but just the bunch of bytes probably not all that useful

File: crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/bootstrap.rs [hunk 4]
```diff
@@ -0,0 +1,41 @@
+lazy_static::lazy_static! {
+    static ref CLIENT: reqwest::Client = reqwest::Client::new();
+}
```
Comment (neonphog) [inline crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/bootstrap.rs:3]:
```suggestion
use once_cell::sync::Lazy;

static CLIENT: Lazy<reqwest::Client> = Lazy::new(|| reqwest::Client::new());
```

once_cell is the new lazy_static - it's more efficient on the back-end.

File: crates/holochain_p2p/src/fixt.rs [hunk 5]
```diff
@@ -56,12 +56,30 @@ fixturator!(
...
     };
     curve Predictable {
-        AgentInfoSigned::try_new(fixt!(KitsuneSignature, Predictable), fixt!(AgentInfo, Predictable)).unwrap()
+        let mut data = Vec::new();
+        kitsune_p2p_types::codec::rmp_encode(&mut data, &fixt!(AgentInfo, Predictable)).unwrap();
+        AgentInfoSigned::try_new(
+            fixt!(KitsuneAgent, Predictable),
+            fixt!(KitsuneSignature, Predictable),
+            data,
+        ).unwrap()
```
Comment (neonphog) [inline crates/holochain_p2p/src/fixt.rs:83]: I'm not sure these make sense as fixturators... the signatures don't match the data, correct?

**Review by neonphog — APPROVED** [review]
couple questions / suggestions - but looks awesome : )

File: crates/holochain_p2p/src/fixt.rs (hunk 5 above)
Comment (thedavidmeister) [reply crates/holochain_p2p/src/fixt.rs:83]: @neonphog yeah the signatures don't match

Comment (thedavidmeister) [reply crates/holochain_p2p/src/fixt.rs:83]: @neonphog the signatures do match now haha

**Pull request merged**